Snowflake.Data/Core/Tools/UnixOperations.cs

```diff
@@ -27,5 +32,24 @@ public virtual bool CheckFileHasAnyOfPermissions(string path, FileAccessPermissi
             var fileInfo = new UnixFileInfo(path);
             return (permissions & fileInfo.FileAccessPermissions) != 0;
         }
+
+        public string ValidateFileAndReadText(string path, Action<UnixStream> validation)
```

Comment:
maybe this method is not needed - we could move its body to the method below

Reply:
Done